src/Arcus.Security.Tests.Integration/HashiCorp/Hosting/HashiCorpVaultTestServer.cs

```diff
@@ -211,11 +237,25 @@ public async Task MountKeyValueAsync(string path, VaultKeyValueSecretEngineVersi
         /// <exception cref="ArgumentNullException">Thrown when the <paramref name="capabilities"/> is <c>null</c>.</exception>
         public async Task AddPolicyAsync(string name, string path, string[] capabilities)
         {
-            Guard.NotNullOrWhitespace(name, nameof(name), "Requires a name to identify the policy");
-            Guard.NotNullOrWhitespace(path, nameof(path), "Requires a path where the policy will be applicable");
-            Guard.NotNull(capabilities, nameof(capabilities), "Requires a set of capabilities that should be available in this policy");
-            Guard.NotAny(capabilities, nameof(capabilities), "Requires a set of capabilities that should be available in this policy");
-            Guard.For<ArgumentException>(() => capabilities.Any(String.IsNullOrWhiteSpace), "Requires all the capabilities of the policy to be filled out (not blank)");
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("Requires a name to identify the policy", nameof(name));
+            }
+
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                throw new ArgumentException("Requires a path where the policy will be applicable", nameof(path));
+            }
+
+            if (capabilities is null || !capabilities.Any())
+            {
+                throw new ArgumentNullException(nameof(capabilities), "Requires a set of capabilities that should be available in this policy");
```

Comment:
This should be an `ArgumentException`, as the previous version also threw an `ArgumentException` instead of an `ArgumentNullException`